### ISSUE TITLE
`@Controller`s cannot use Guards as it throws a `Not found module` error

### ISSUE THREAD

**Issue by V-ed** [fjodor-rybakov/discord-nestjs#709]
**Describe the bug**
When using Controllers, adding any guard in a `@UseGuards` of either a class scope or a method scope gives this error :

```
Error: Not found module for InteractionsGateway class
    at InstantiationService.getHostModule (<project path>/node_modules/.pnpm/@discord-nestjs+core@3.0.6_650a03e96f43d1bb974a6fd03a710d01/node_modules/@discord-nestjs/core/src/services/instantiation.service.ts:19:13)
    at GuardResolver.<anonymous> (<project path>/node_modules/.pnpm/@discord-nestjs+core@3.0.6_650a03e96f43d1bb974a6fd03a710d01/node_modules/@discord-nestjs/core/src/resolvers/guard/guard.resolver.ts:48:50)
    at Generator.next (<anonymous>)
    at <project path>/node_modules/.pnpm/@discord-nestjs+core@3.0.6_650a03e96f43d1bb974a6fd03a710d01/node_modules/@discord-nestjs/core/dist/resolvers/guard/guard.resolver.js:17:71
    at new Promise (<anonymous>)
    at __awaiter (<project path>/node_modules/.pnpm/@discord-nestjs+core@3.0.6_650a03e96f43d1bb974a6fd03a710d01/node_modules/@discord-nestjs/core/dist/resolvers/guard/guard.resolver.js:13:12)
    at GuardResolver.resolve (<project path>/node_modules/.pnpm/@discord-nestjs+core@3.0.6_650a03e96f43d1bb974a6fd03a710d01/node_modules/@discord-nestjs/core/dist/resolvers/guard/guard.resolver.js:62:16)
    at DiscordResolverService.<anonymous> (<project path>/node_modules/.pnpm/@discord-nestjs+core@3.0.6_650a03e96f43d1bb974a6fd03a710d01/node_modules/@discord-nestjs/core/src/services/discord-resolver.service.ts:97:36)
    at Generator.next (<anonymous>)
    at fulfilled (<project path>/node_modules/.pnpm/@discord-nestjs+core@3.0.6_650a03e96f43d1bb974a6fd03a710d01/node_modules/@discord-nestjs/core/dist/services/discord-resolver.service.js:14:58)
```

**To Reproduce**
Create a Gateway / controller with such kind of setup :

```ts
import { UseGuards } from "@discord-nestjs/core";
import { Controller } from "@nestjs/common";
import { RandomGuard } from "./my-random.guard";

@Controller()
@UseGuards(RandomGuard)
export class InteractionsGateway {}
```

and you should see the issue.

Note that a Guard is needed in the parameters of `@UseGuards`, as an empty argument (`@UseGuards()`) will still work.

**Expected behavior**
Controllers work like they used to in versions < 3.

**Version**
3.0.6

**Additional context**
A workaround is to not create Controller (and adding those controllers to the `controllers` property of any module) and instead use `@Injectable` and add the gateway to the `providers` list of any module. This makes the gateway work and guards are also applied.

**Comment by fjodor-rybakov** [comment]
Hi! I added potential fix in version 3.0.7

**Comment by V-ed** [comment]
Well, that was much faster than I anticipated, thanks :D

Version 3.0.7 seems to indeed fix this issue, thanks!
